TYPO3.Neos/Resources/Public/JavaScript/Content/Navigate/AbstractNodeTree.js

```diff
@@ -866,7 +866,18 @@ define(
 							// Update the pageNodePath if we moved the current page
 							if (that.get('pageNodePath') === sourceNode.data.key) {
 								that.set('pageNodePath', result.data.newNodePath);
+							} else {
+								// handle pageNodePath if we moved a parent node
+								var explodedParentPath = sourceNode.data.key.split("@");
+
+								if(that.get('pageNodePath').indexOf(explodedParentPath[0]) === 0 ) {
+									var newExplodedPath = result.data.newNodePath.split("@");
+									var currentNodePath = that.get('pageNodePath').substr(explodedParentPath[0].length);
+
+									that.set('pageNodePath', newExplodedPath[0] + currentNodePath);
```

Comment:
``that.set('pageNodePath', that.get('pageNodePath').replace(explodedParentPath[0], newExplodedPath[0]));`` might be easier to comprehend